### PR TITLE
Update server_updateObject.sqf

### PR DESCRIPTION
**Pull request by Mikeeeyy**
Better formatting and some fixes, one major fix involving the saving of inventory. It was using format which has a limit. That could be a problem with a vehicle that can hold a lot of cargo with a ton of unique mags/weapons in it.

### CODE REVIEW

File: SQF/dayz_server/compile/server_updateObject.sqf [hunk 1]
```diff
@@ -18,173 +20,163 @@ _firstTime = false;
...
     _uid = "0";
 };
-if (!_parachuteWest && !(locked _object)) then {
-	if (_objectID == "0" && _uid == "0") then
-	{
+
+if (!_parachuteWest && !locked _object) then {
+	if (_objectID == "0" && _uid == "0") then {
 		_object_position = getPosATL _object;
     	_isNotOk = true;
```
Comment (Namindu) [inline SQF/dayz_server/compile/server_updateObject.sqf]: Could use an indent forwards

File: SQF/dayz_server/compile/server_updateObject.sqf [hunk 2]
```diff
@@ -18,173 +20,163 @@ _firstTime = false;
...
 	} count _hitpoints;
-	
-	if (_objectID == "0") then {
-		_key = format["CHILD:306:%1:%2:%3:",_uid,_array,_damage];
-	} else {
-		_key = format["CHILD:306:%1:%2:%3:",_objectID,_array,_damage];
-	};
-	//diag_log ("HIVE: WRITE: "+ str(_key));
+
+	_key = "CHILD:306:" + _objectID + ":" + str _array + ":" + str _damage;
```
Comment (ebaydayz) [inline SQF/dayz_server/compile/server_updateObject.sqf]: Why did you take out the if () else here? _objectID may still equal zero right?

File: SQF/dayz_server/compile/server_updateObject.sqf [hunk 3]
```diff
@@ -18,173 +20,163 @@ _firstTime = false;
...
+	{
+		_hit = [_object, _x] call object_getHit;
+		_selection = getText (configFile >> "CfgVehicles" >> typeOf _object >> "HitPoints" >> _x >> "name");
+		if (_hit > 0) then {
+			_array set [count _array, [_selection,_hit]];
+		};
+		_object setHit [_selection, _hit];
+	} count _hitpoints;
+
+	_key = "CHILD:306:" + _objectID + ":" + str _array + ":" + str _damage;
```
Comment (ebaydayz) [inline SQF/dayz_server/compile/server_updateObject.sqf]: Forgot the last ":" after str _damage. Should be ... + str _damage + ":";

File: SQF/dayz_server/compile/server_updateObject.sqf [hunk 4]
```diff
@@ -18,173 +20,163 @@ _firstTime = false;
...
-		};
+	_inventory = [
+		getWeaponCargo _object,
+		getMagazineCargo _object,
+		getBackpackCargo _object
+	];
+	_previous = str (_object getVariable ["lastInventory", []]);
+	if (str _inventory != _previous) then {
+		_object setVariable ["lastInventory", _inventory];
+		_key = if (_objectID == "0") then { format ["CHILD:309:%1:", _uid] + str _inventory } else { format ["CHILD:303:%1:", _objectID] + str _inventory };
```
Comment (ebaydayz) [inline SQF/dayz_server/compile/server_updateObject.sqf]: Forgot the last ":" should be ... + str _inventory + ":"

File: SQF/dayz_server/compile/server_updateObject.sqf [hunk 5]
```diff
@@ -18,173 +20,163 @@ _firstTime = false;
...
+		if (_hit > 0) then {
+			_array set [count _array, [_selection, _hit]];
+		};
 		_object setHit [_selection, _hit];
 	} count _hitpoints;
-	
-	_key = format["CHILD:306:%1:%2:%3:",_objectID,_array,_damage];
-	//diag_log ("HIVE: WRITE: "+ str(_key));
+
+	_key = "CHILD:306:" + _objectID + ":" + str _array + ":" + str _damage;
```
Comment (ebaydayz) [inline SQF/dayz_server/compile/server_updateObject.sqf]: Forgot the last ":"

**Comment by ebaydayz** [comment]
Not sure if all of those variables you switched to use str () instead of format [] actually reach the 2048 character limit, but it is probably a good idea to do it anyway. Better safe than sorry. :+1:

File: SQF/dayz_server/compile/server_updateObject.sqf (hunk 2 above)
Comment (Mikeeeyy) [inline SQF/dayz_server/compile/server_updateObject.sqf]: If you look it's using the same 306 call for the objectID and objectUID which isn't correct. Vehicles also only use the objectID, they don't use objectUID.

File: SQF/dayz_server/compile/server_updateObject.sqf (hunk 3 above)
Comment (Mikeeeyy) [inline SQF/dayz_server/compile/server_updateObject.sqf]: My mistake, I use extDB you see.

**Comment by Mikeeeyy** [comment]
Only the inventory needed it tbh, and maybe the hitpoints array. Thought I may as well do the rest while I was at it.

**Pull request merged**